File: Classes/C++/Product.cpp
#include "Product.h"

Product::Product(int id, string name, string sku, float price, string category) {
    this->id       = id;
    this->name     = name;
    this->sku      = sku;
    this->price    = price;
    this->category = category;
}

Product::~Product() {}

// Getters
int    Product::getId()       { return id; }
string Product::getName()     { return name; }
string Product::getSku()      { return sku; }
float  Product::getPrice()    { return price; }
string Product::getCategory() { return category; }

// Setters
void Product::setName(string name)         { this->name     = name; }
void Product::setSku(string sku)           { this->sku      = sku; }
void Product::setPrice(float price)        { this->price    = price; }
void Product::setCategory(string category) { this->category = category; }

File: classes/Product.ts
import { ProductDef } from "./ProductDef";

export class Product extends ProductDef {
  constructor(id: number, name: string, sku: string, price: number, category: string) {
    super();
    this.setId(id);
    this.setName(name);
    this.setSku(sku);
    this.setPrice(price);
    this.setCategory(category);
  }

  getId(): number       { return this.id; }
  getName(): string     { return this.name; }
  getSku(): string      { return this.sku; }
  getPrice(): number    { return this.price; }
  getCategory(): string { return this.category; }

  setId(id: number): void             { this.id       = id; }
  setName(name: string): void         { this.name     = name; }
  setSku(sku: string): void           { this.sku      = sku; }
  setPrice(price: number): void       { this.price    = price; }
  setCategory(category: string): void { this.category = category; }
}
